src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AwsLambda/HandlerMethodWrapper.cs

```diff
@@ -36,46 +36,102 @@ public AfterWrappedMethodDelegate BeforeWrappedMethod(InstrumentedMethodCall ins
         {
             var isAsync = instrumentedMethodCall.IsAsync;
 
-            var inputObject = instrumentedMethodCall.MethodCall.MethodArguments[0];
-            var lambdaContext = instrumentedMethodCall.MethodCall.MethodArguments[1]; // TODO handle case where this doesn't exist
+            object lambdaContext = null;
+            object inputObject = null;
+            AwsLambdaEventType eventType = AwsLambdaEventType.Unknown;
 
-            var functionNameGetter = _getFunctionNameFromLambdaContext ??= VisibilityBypasser.Instance.GeneratePropertyAccessor<string>(lambdaContext.GetType(), "FunctionName");
-            var functionVersionGetter = _getFunctionVersionFromLambdaContext ??= VisibilityBypasser.Instance.GeneratePropertyAccessor<string>(lambdaContext.GetType(), "FunctionVersion");
-            var requestIdGetter = _getAwsRequestIdFromLambdaContext ??= VisibilityBypasser.Instance.GeneratePropertyAccessor<string>(lambdaContext.GetType(), "AwsRequestId");
-            var functionArnGetter = _getInvokedFunctionArnFromLambdaContext ??= VisibilityBypasser.Instance.GeneratePropertyAccessor<string>(lambdaContext.GetType(), "InvokedFunctionArn");
+            foreach (var arg in instrumentedMethodCall.MethodCall.MethodArguments)
+            {
+                // TODO: I don't like this approach, but our options are limited without access to the interface
```

Comment:
One improvement we could make is to only do some of this work on the initial invocation, because some of this data should not change with each invocation. That could be done in a separate PR if we think it is worthwhile.

Reply:
I think you're probably right but I'd like to see more real data to make sure they can't actually change on the fly. :)

Reply:
I ended up doing as you suggested. I realized we would have to handle more race conditions if we expect some of the values to change.